### PR TITLE
fix: email configuration checks to account for sendgrid

### PR DESCRIPTION
**Pull request by emil2k**
Checklist:

* [ ] Any new values are backwards compatible and/or have sensible default.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/kubeshop/testkube-cloud-charts/blob/master/community/CONTRIBUTING.md).
* [ ] My CI is green.

### CODE REVIEW

File: charts/testkube-cloud-api/values.yaml [hunk 1]
```diff
@@ -145,21 +145,26 @@ api:
...
-  #   # -- SMTP username
-  #   username: ""
-  #   # -- SMTP password
-  #   password: ""
-  #   # -- SMTP secret ref (secret must contain key SMTP_PASSWORD), overrides password field if defined
-  #    passwordSecretRef: ""
+  inviteMode: auto-accept
+  email:
+    fromEmail: ""
+    fromName: ""
```
Comment (dejanzele) [inline charts/testkube-cloud-api/values.yaml:151]: can you add comments on those two fields also and regenerate helm docs?

File: charts/testkube-enterprise/values.yaml [hunk 2]
```diff
@@ -249,21 +249,26 @@ testkube-cloud-api:
...
-    #     # -- SMTP username
-    #     username: ""
-    #     # -- SMTP password
-    #     password: ""
-    #     # -- SMTP secret ref (secret must contain key SMTP_PASSWORD), overrides password field if defined
-    #      passwordSecretRef: ""
+    inviteMode: auto-accept
+    email:
+      fromEmail: ""
+      fromName: ""
```
Comment (dejanzele) [inline charts/testkube-enterprise/values.yaml:255]: can you add comments and regenerate helm docs?

File: charts/testkube-enterprise/values.yaml [hunk 3]
```diff
@@ -249,21 +249,26 @@ testkube-cloud-api:
...
+      # -- SMTP username
+      username: ""
+      # -- SMTP password
+      password: ""
+      # -- SMTP secret ref (secret must contain key SMTP_PASSWORD), overrides password field if defined
+      passwordSecretRef: ""
+    sendgrid:
+      # -- Sendgrid API key
+      apiKey: ""
+      # -- Secret API key secret ref (secret must contain key SENDGRID_API_KEY) (default is `sendgrid-api-key`)
```
Comment (dejanzele) [inline charts/testkube-enterprise/values.yaml:270]:
```suggestion
      # -- Secret API key secret ref (secret must contain key SENDGRID_API_KEY)
```

File: charts/testkube-cloud-api/templates/deployment.yaml [hunk 4]
```diff
@@ -88,6 +88,7 @@ spec:
               value: "{{ .Values.api.inviteMode }}"
             {{- if eq .Values.api.inviteMode "email" }}
             {{- $check := required ".Values.api.email needs to be configured with .Values.api.inviteMode set to \"email\"" .Values.api.email -}}
+            {{- if not (or .Values.api.sendgrid.secretRef .Values.api.sendgrid.apiKey) }}
```
Comment (ypoplavs) [inline charts/testkube-cloud-api/templates/deployment.yaml:91]: hm, is this condition needed?

Comment (emil2k) [reply charts/testkube-cloud-api/templates/deployment.yaml:91]: yes, so it does not run the checks on the smtp values as we will be using sendgrid instead.

**Review by ypoplavs — APPROVED** [review]

**Review by dejanzele — APPROVED** [review]

**Pull request merged**